server/NucleiSegmentation/NucleiSegmentation.py

```diff
@@ -17,15 +18,33 @@
 }
 
 
+def read_image(file):
+    return skimage.io.imread(file)[:, :, :3]
+
+
 def main(args):
 
     #
     # Read Input Image
     #
     print('>> Reading input image')
 
-    imInput = skimage.io.imread(args.inputImageFile)[:, :, :3]
-
+    imInput = None
+    imageFile = args.inputImageFile
+    if os.path.isdir(imageFile):
+        dir = imageFile
+        for file in os.listdir(dir):
```

Comment:
@jbeezley Why allow it to be a directory, loop through all files, try loading each one, and break at the first one that succeeds.

The front-end you developed already includes a nice widget to select an image file from any folder on girder. If the user selects a wrong format, we should maybe let the job fail. If you want to restrict input image files to simple png/jpg, we can add the fileExtenstions attribute like [this](https://github.com/DigitalSlideArchive/HistomicsTK/blob/master/server/NucleiSegmentation/NucleiSegmentation.xml#L28) for the inputImageFile parameter. Then your file selection GUI can restrict it to whichever file formats are mentioned in that list.